Core/include/Acts/Geometry/ProtoDetector.hpp

```diff
@@ -0,0 +1,106 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/Extent.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Utilities/BinningData.hpp"
+
+#include <string>
+
+namespace Acts {
+
+/// A proto volume description being used to define an overall
+/// structure of either a TrackingVolume or Experimental::DetectorVolume
+struct ProtoVolume {
+  /// Name of the proto volume
+  std::string name = "";
+  /// The extent of this layer
+  Extent extent;
+  /// Boolean to indicate this is a container, needed for legacy
+  bool layerContainer = false;
+
+  /// Set the layer type to indicate the layer volume
+  Acts::Surface::SurfaceType layerType = Acts::Surface::SurfaceType::Other;
+  /// The surface binninng fo the layer
+  std::vector<BinningData> layerSurfaceBinning = {};
```

Comment:
should we bundle all the legacy information into a legacy struct?

e.g.

```
struct Legacy {
  /// Boolean to indicate this is a container, needed for legacy
  bool layerContainer = false;
  /// Set the layer type to indicate the layer volume
  Acts::Surface::SurfaceType layerType = Acts::Surface::SurfaceType::Other;
  /// The surface binninng fo the layer
  std::vector<BinningData> layerSurfaceBinning = {};

  // ...
}

std::optional<Legacy> legacy;
```

Reply:
Oh, that's a good suggestion.

Reply:
Actually, it only needs a `legacyLayerType`, the other information is either both legacy or new, or redundant.